src/library/searchqueryparser.cpp

```diff
@@ -177,8 +177,17 @@ void SearchQueryParser::parseTokens(QStringList tokens,
             }
             // Don't trigger on a lone minus sign.
             if (!token.isEmpty()) {
-                pNode = std::make_unique<TextFilterNode>(
-                                m_pTrackCollection->database(), searchColumns, token);
+                // For untagged strings we search the track fields as well
+                // as the crate names the track is in. This allows the user
+                // to use crates like tags
+                std::unique_ptr<OrNode> gNode = std::make_unique<OrNode>();
```

Comment:
What does the prefix "g" stand for? Group? Why do we need to introduce another temporary here instead of just storing the OrNode in pNode?

Reply:
pNode is not of GroupNode subtype and will therefor not have addNode. At least I was not able to get it compiling, maybe with 2 casts, but then I would argue it looks much better with a second variable.

Reply:
✔️